### PR TITLE
vulnfeeds: Improve accuracy of PyPI package matching.

### PR DESCRIPTION
**Pull request by oliverchang**
Try to parse additional URLS for PyPI package descriptions and use that
to cross-reference against CVEs.

### CODE REVIEW

File: vulnfeeds/pypi/pypi.sql [hunk 1]
```diff
@@ -18,10 +18,34 @@ CREATE TEMP FUNCTION PROCESS_LINKS(val ARRAY<STRING>) AS ((
...
+CREATE TEMP FUNCTION EXTRACT_LINKS(name STRING, description STRING)
+RETURNS ARRAY<STRING>
+LANGUAGE js
+AS r"""
+if (!description) {
+  return [];
+}
+
+let results = [];
+for (let link of description.matchAll(/https:\/\/[A-Za-z0-9.\/#?&@=_\-]+/g)) {
```
Comment (inferno-chromium) [inline vulnfeeds/pypi/pypi.sql:31]: s/link/links ?

File: vulnfeeds/pypi/pypi.sql [hunk 2]
```diff
@@ -18,10 +18,34 @@ CREATE TEMP FUNCTION PROCESS_LINKS(val ARRAY<STRING>) AS ((
...
+  link = link[0];
+  if (link.toLowerCase().includes(name.toLowerCase())) {
+    results.push(link.replace(/\.$/, ''));
```
Comment (inferno-chromium) [inline vulnfeeds/pypi/pypi.sql:34]: why this?

**Review by inferno-chromium — APPROVED** [review]

**Review by jonathanmetzman — APPROVED** [review]
LGTM

File: vulnfeeds/pypi/pypi.sql (hunk 1 above)
Comment (oliverchang) [reply vulnfeeds/pypi/pypi.sql:31]: This is iterating one link at a time.

File: vulnfeeds/pypi/pypi.sql (hunk 2 above)
Comment (oliverchang) [reply vulnfeeds/pypi/pypi.sql:34]: Added a comment here.

**Pull request merged**